### PR TITLE
Improve project contributing guidelines

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

Addressed comments in made in #6595:

> Combine the contribute page and the contributing page to have one location

I merged both documents and enhanced the docs build process to copy the main contributing guidelines into the docs page.

> Provide some details on what to do if the tests npm run test do not pass after a clean install and setup

Added a comment that if tests don't pass, people should create an issue.

> Emphasise using the examples over npm link and provide instructions for how the examples should be run, especially for the setup of a browser driver and/or some instructions on how to use the WebDriver Mock Service.

Removed the linking part completely and extended docs on the examples within the repo.

> Links to some decent Typescript resources

Added links to [awesome typescript essential resources](https://github.com/dzharii/awesome-typescript#awesome-typescript-essential-resources).

> For the documentation, are there any writing style guides? Note that in some cases the same data may be in multiple places e.g. in examples files.

Added a note to that as well.

> A primer on the structure of the smoke tests

Extended existing documentation on smoke tests with explanation on its structure.

> A primer on the mocking used in the Jest tests and general structure

Added a primer on unit testing in general with information where tests are located and how to mock.

> A walk through of the pull request process. Do I resolve conversations when the issue has been fixed or does the reviewer?

Due to the merge of both contributing docs this is now part of the one document we have.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

cc @RossVertizan

### CODE REVIEW

**Comment by RossVertizan** [comment]
That was quick - thank you.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,10 +1,54 @@
...
+
+## Make a Pull Request
+
+Once you have a fix implemented or finished a feature implementation you can make a pull request. Your changes needs to be pushed on your WebdriverIO fork. In the GitHub UI you should see a button popping up that allows you to raise a PR to the main repository.
+
+We already provide a template for you to fill out. There are not many rules to follow here. Just try to explain your change with as much detail as possible. Make sure that you have written enough unit tests for your changes otherwise the code coverage check will let the build fail.
+
+Like in many Open Source projects we ask you to sign a __CLA__ which is a Contributor License Agreement that ensures that all contributions to the project are licensed under the project's respective open source license, which is MIT. It regulates the legal implications of you providing us (as the OpenJS Foundation) code changes.
```
Comment (dylanlive) [inline CONTRIBUTING.md:49]: Is it possible to link to the CLA text here?

Comment (christian-bromann) [reply CONTRIBUTING.md:49]: Unfortunately every PR has its own link to a CLA. There is no general CLA afaik.

Comment (dylanlive) [reply CONTRIBUTING.md:49]: Okay, I kind of figured. It was the reason it took me weeks to get my CLA signed (after opening the PR) as it had to be reviewed. Would have loved to have had it reviewable prior to opening a PR.

**Review by WillBrock — APPROVED** [review]
LGTM

**Pull request merged**